### ISSUE TITLE
Indent a multi-line query

### ISSUE THREAD

**Issue by amjith** [dbcli/pgcli#184]
In multi-line mode the query should be indented by a leading 4 spaces and a `->` prompt.

**Comment by jonathanslenders** [comment]
I'll do that.

**Comment by jonathanslenders** [comment]
Currently, when a user does a reverse-i-search (Control-R) or a repetition (meta+number), this is shown before the input instead of the actual prompt.  You get ``(reverse-i-search `text`): `` or ``(arg: 3)`` for instance. This means that while doing this, the width of the prompt changes. That is fine for single line input, but I noticed that if we indent all of the lines, this can become anoying, because long lines refloat.

For ``ptpython``, I solved that by showing a complete separate ``I-Search: `` and ``Arg: `` prompt at the bottom. Would you agree to take the same approach as ptpython. I would implement this as an alternative for ``prompt_toolkit.shortcuts.create_default_layout``.

So, the layout engine of ``prompt_toolkit`` supports all possibilities, but in the ``shortcuts`` module I try to expose only one or two sane defaults that are well tested. This will become the second default; a better multiline friendly one.

**Comment by amjith** [comment]
Yeah I think moving the search bar down is a reasonable solution. 

Thanks again for addressing this.

**Comment by jonathanslenders** [comment]
Hi, indentation is working in the master branch of prompt-toolkit. All the examples I tested seem to work, but I'm probably going to give it a few more days before a release.

![pgsql](https://cloud.githubusercontent.com/assets/216638/9071081/7eeba2f6-3af3-11e5-974f-45a59ce38e20.png)


The only change for pgcli:
```diff
diff --git a/pgcli/main.py b/pgcli/main.py
index 29ddddd..5b8d50b 100755
--- a/pgcli/main.py
+++ b/pgcli/main.py
@@ -241,6 +241,7 @@ class PGCli(object):
                                        reserve_space_for_menu=True,
                                        get_prompt_tokens=prompt_tokens,
                                        get_bottom_toolbar_tokens=get_toolbar_tokens,
+                                       multiline=True,
                                        extra_input_processors=[
                                            # Highlight matching brackets while editing.
                                            ConditionalProcessor(
```
However, I'd like to use another name instead of "multiline" here... It's a feature that reflows lines with indentation and shows search and arg toolbars at the bottom.

**Comment by amjith** [comment]
I think it's fine to call it a different name in prompt_toolkit but I think it should still be called multi-line mode in pgcli/mycli. 

Because as far as the user is concerned they have enabled a mode that lets them write queries that span multiple lines. It just so happens the lines are indented by default now.

**Comment by amjith** [comment]
Yay! Pasting works as expected without adding extra indentation. :)

**Comment by amjith** [comment]
This is now available in v1.0. Please upgrade. 

```
pip install -U pgcli
```
